### ISSUE TITLE
An error occurs on Node 18.12.1.

### ISSUE THREAD

**Issue by gmjung-platform** [colyseus/colyseus#545]
The following error occurs on Node 18.12.1.

Error: This version of µWS is not compatible with your Node.js build:

Error: Cannot find module './uws_darwin_arm64_108.node'
Require stack:

/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/uWebSockets.js/uws.js
/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/@colyseus/uwebsockets-transport/build/uWebSocketsTransport.js
/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/@colyseus/uwebsockets-transport/build/index.js
/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/dist/Server.js
at /Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/uWebSockets.js/uws.js:22:9
at Object. (/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/uWebSockets.js/uws.js:24:3)
at Module._compile (node:internal/modules/cjs/loader:1159:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
at Module.load (node:internal/modules/cjs/loader:1037:32)
at Module._load (node:internal/modules/cjs/loader:878:12)
at Module.require (node:internal/modules/cjs/loader:1061:19)
at require (node:internal/modules/cjs/helpers:103:18)
at Object. (/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/@colyseus/uwebsockets-transport/build/uWebSocketsTransport.js:6:19)
at Module._compile (node:internal/modules/cjs/loader:1159:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
at Module.load (node:internal/modules/cjs/loader:1037:32)
at Module._load (node:internal/modules/cjs/loader:878:12)
at Module.require (node:internal/modules/cjs/loader:1061:19)
at require (node:internal/modules/cjs/helpers:103:18)
at Object. (/Volumes/Workspace/Developments/workspace_for_game/game-server-with-node.js/node_modules/@colyseus/uwebsockets-transport/build/index.js:6:28)

**Comment by goldsziggy** [comment]
This happened to me as well -- It looks like colyseus brings in `uWebSockets.js` [1] pinned at version 19.2.0.  According to uWebSockets.js releases NodeJS 18 wasn't supported till v20.7.0.  This upgrade comes with a downside of no longer supporting non-LTS NodeJS releases (not necessarily a bad thing long term - but may need documentation).

One side-effect of the upgrade is; users of NodeJS 15 will no longer be able to update; although this isn't necessarily bad as NodeJS 15 went out of support 1.5 years ago.  NodeJS 16 only has ~6 months left of support as well so I think this package should really be updated.


[1]: https://github.com/colyseus/colyseus/blob/master/packages/transport/uwebsockets-transport/package.json
[2]: https://github.com/uNetworking/uWebSockets.js/releases/tag/v20.7.0

**Comment by jerricko** [comment]
As a workaround, you can override the uWebSockets version in package.json. I've had mine set to v20 for awhile and everything has been working so far on node v18.

In dependencies, you can do:
```
"uWebSockets.js": "github:uNetworking/uWebSockets.js#v20.10.0"
```
and then add an override (place below dependencies):
```
"overrides": {
    "@colyseus/uwebsockets-transport": {
      "uWebSockets.js": "$uWebSockets.js"
    }
  }
```

Set the version to whatever you want.

**Comment by endel** [comment]
Thanks for the workaround @jerricko, that's really helpful!

Closing as fixed on 0.15
